### ISSUE TITLE
Digger reports success even though Terraform apply fails

### ISSUE THREAD

**Issue by peteher** [diggerhq/digger#1658]
### Problem:
Terraform apply reports error/exit code 1, but pipeline won't fail as Digger saying "success"

### Setup:
- no-backend setup
- terraform 1.5.7
- digger vLatest today - v0.6.33
- upload-plan-destination: github
- digger.yaml:
```
projects:
  - name: default
    workflow: my_custom_workflow
workflows:
  my_custom_workflow:
    workflow_configuration:
      on_pull_request_pushed: [ digger plan ]
      on_pull_request_closed: [ digger unlock ]
      on_commit_to_default: [ digger apply ]
auto_merge: false
collect_usage_data: false
```
### Flow:
1. changes delivered to branch, PR (ie tag resources)
2. action runs to "plan" for PR
3. plan.out attached to action
4. make change on infra via console for eg delete a resource that the plan wants to tag
5. merge PR to "apply"
6. Terraform apply will fail as step 1 change can't be committed on step 4 removed resource
7. Digger would say "apply failed, skipping job" and "success" - pipeline finishing green ✅ - I'd expect ❌ 

![image](https://github.com/user-attachments/assets/dc24057b-b9b3-4c2f-b5bc-a819ecae7a54)

Using version v0.3.22 I get the expected fail:
![image](https://github.com/user-attachments/assets/098ae236-e7e3-42f6-a25e-dcab7ab7cc05)

**Comment by motatoes** [comment]
Hi @peteher thanks for reporting this, might be related to https://github.com/diggerhq/digger/pull/1579 as well

But it is backendless so I believe this is a slightly different issue for run Jobs

**Comment by Kamalesh-Seervi** [comment]
can I work on this issue or its solved

**Comment by fleroux514** [comment]
Just noticed this issue in v0.6.44.

To me this is a blocker since it would allow someone to merge a PR even tough the apply is failing. Preventing this use case is the core purpose of digger so I don't understand that it hasn't been addressed since v0.6.33.

It would be nice if there would be a `blocker` tag for issues that must be fixed in upcoming releases, before even thinking of adding new functionality. Bugs in new functionality is understandable but regressions are not acceptable.

**Comment by motatoes** [comment]
Hi folks just realised this issue made it to the top upvotes, I didn't realise it was an issue of high severity. Thanks @fleroux514 for the blocker idea I think that will adopt this strategy. I will try to take a look at this soon

**Comment by motatoes** [comment]
For some reason when I read this issue previously I thought it was related to the one I linked of posting the error back to the comment, did not realise that the pipeline was succeeding

**Comment by minamijoyo** [comment]
@motatoes This issue is a really blocker for me using Digger in production, so I fixed it in #1760. Please review 🙏

**Comment by minamijoyo** [comment]
Fixed in v0.6.50 🚀
